bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Server.java

```diff
@@ -80,7 +81,8 @@ private Server(String[] cmdargs, Properties systemProps,
         this.systemEnv = systemEnv;
         this.moduleLoader = moduleLoader;
         this.shutdownHandler = shutdownHandler;
-        this.clientFactorySvcCaptureRef = new AtomicReference<>(null);
+        this.clientFactorySvcCaptureRef = new AtomicReference<>();
+        this.notifierRef = new AtomicReference<>();
```

Comment:
This shouldn't need to be a field, as its use is confined to one method and the anonymous Service impl created by a call from that method.